src/test/ui/proc-macro/issue-39889.rs

```diff
@@ -1,6 +1,6 @@
 // run-pass
 
-#![allow(dead_code)]
+#![allow(dead_code, unused_macros)]
```

Comment:
This one is very insteresting.

https://github.com/rust-lang/rust/blob/493ea776ebdecb7ab318444525115810de255c20/src/test/ui/proc-macro/auxiliary/issue-39889.rs#L10-L17

Any idea why the unused macro wasn't linted beforehand?

Reply:
Good question, no idea. My best guess is that now we have a span for `id`, but we didn't for the whole macro. I think changing the test is reasonable. [The issue is about the macro expansion machinery](https://github.com/rust-lang/rust/issues/39889), so it's reasonable to `allow` it.